### PR TITLE
Support both Python 3.6 and Pydantic 1.10

### PR DESCRIPTION
**Pull request by svlandeg**
Adding support for Pydantic 1.10.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -9,7 +9,8 @@ catalogue>=2.0.4,<2.1.0
 confection>=0.0.1,<1.0.0
 ml_datasets>=0.2.0,<0.3.0
 # Third-party dependencies
-pydantic>=1.7.4,!=1.8,!=1.8.1,<1.10.0
+pydantic>=1.7.4,!=1.8,!=1.8.1,<1.10.0; python_version < "3.7"
```
Comment (polm) [inline requirements.txt:12]: I was confused by the specifically excluded versions here, but it looks like Pydantic has a `1.8` but no `1.8.0`. Just commenting for anyone else unfamiliar with this.

File: requirements.txt [hunk 2]
```diff
@@ -9,7 +9,8 @@ catalogue>=2.0.4,<2.1.0
...
+pydantic>=1.7.4,!=1.8,!=1.8.1,<1.11.0; python_version >= "3.7"
```
Comment (adrianeboyd) [inline requirements.txt:13]: You don't need the split requirements, pip just won't install newer versions if they're marked as python 3.7+.

```suggestion
pydantic>=1.7.4,!=1.8,!=1.8.1,<1.11.0
```

File: setup.cfg [hunk 3]
```diff
@@ -49,7 +49,8 @@ install_requires =
     # Third-party dependencies
     setuptools
     numpy>=1.15.0
-    pydantic>=1.7.4,!=1.8,!=1.8.1,<1.10.0
+    pydantic>=1.7.4,!=1.8,!=1.8.1,<1.10.0; python_version < "3.7"
+    pydantic>=1.7.4,!=1.8,!=1.8.1,<1.11.0; python_version >= "3.7"
```
Comment (adrianeboyd) [inline setup.cfg:53]:
```suggestion
    pydantic>=1.7.4,!=1.8,!=1.8.1,<1.11.0
```

**Comment by svlandeg** [comment]
@adrianeboyd : honestly, I'm not sure where my brain went to have the split requirements. Haha. Anyway, thanks. Merging 🎉

**Pull request merged**